taipy/gui_core/_context.py

```diff
@@ -502,10 +504,10 @@ def crud_scenario(self, state: State, id: str, payload: t.Dict[str, str]):  # no
                 state.assign(error_var, f"Error creating Scenario. {e}")
             finally:
                 self.scenario_refresh(scenario_id)
-                if scenario and (sel_scenario_var := args[1] if isinstance(args[1], str) else None):
+                if (scenario or user_scenario) and (sel_scenario_var := args[1] if isinstance(args[1], str) else None):
                     try:
                         var_name, _ = gui._get_real_var_name(sel_scenario_var)
-                        state.assign(var_name, scenario)
+                        state.assign(var_name, scenario or user_scenario)
```

Comment:
Is there a risk this raises an exception?

Reply:
the risk is that the variable name is not found in the default context